skrifa/src/metrics.rs

```diff
@@ -308,7 +308,10 @@ impl<'a> GlyphMetrics<'a> {
         } else if self.gvar.is_some() {
             advance += self.metric_deltas_from_gvar(glyph_id)[1];
         }
-        Some(self.fixed_scale.apply(advance))
+        // FT casts advance to FT_UShort after applying variation delta and
```

Comment:
For which cases is this FT emulation needed, and do we still provide correct behavior? Does it lead to an intentional truncation? 

Background: We are just discussing a new COLRv1 lack of precision issues due to FT's limited internal representation around variation deltas:
https://gitlab.freedesktop.org/freetype/freetype/-/issues/1257
See also
https://gitlab.freedesktop.org/freetype/freetype/-/commit/1178227b39a4eeecbefad226d4a753b345a16eb1

Reply:
This was actually triggered by a case where FT wraps a negative value. This one is difficult to accommodate in testing because the scale is applied after the fact. 

And yes, I’ve been tracking that issue and discussed it with Rod yesterday. We agreed that COLR seems like a good place to experiment with breaking compatibility for more accurate results so we’d be fine with making those changes here.

Reply:
The issue I was quoting from FT originates from COLRv1 but applies to all variation values (MVAR, HVAR, etc., [listed in the issue](https://gitlab.freedesktop.org/freetype/freetype/-/issues/1257#note_2114118)) that use `tt_var_get_item_delta` - is that where the truncations happens? 

What kind of value was that that was wrapped? This really sounds like a bug in FT - I lean towards not emulating that.

Reply:
Further thinking: If that issue in FT was fixed (for example running FT with @tursunova's patch (which still needs a bit of work, but should cover most cases)) - would then the fauntlet runs and FT match with regards to this line?

Reply:
That patch doesn’t seem to touch the unsigned short cast so we’d still have a diff. This case might be a buggy font because it generates a negative advance width. I think those are generally unexpected in text layout. 

On the broader question, we do FT style truncation for deltas at the call site rather than in the IVS code so we can adjust as needed. We do also have the rounding though, which I ported from your 64 bit accumulation patch.

Reply:
> which I ported from your 64 bit accumulation patch.

Which one do you mean, or which point in the code in FT is that?

> On the broader question, we do FT style truncation for deltas at the call site rather than in the IVS code so we can adjust as needed. 

I have trouble understanding that sentence. What do you mean by "adjust as needed" in this context?

Reply:
That cast occurs in `tt_face_get_metrics` which returns the advance through a pointer to `FT_UShort`.

Reply:
I meant the rounding in FT_MulAddFix here: https://gitlab.freedesktop.org/freetype/freetype/-/blob/master/src/base/ftcalc.c?ref_type=heads#L1083

wrt call site vs IVS, we don’t have any generic “item delta” function that does truncation. Wherever we are applying a delta we truncate explicitly, usually with a comment that this is done to match FT. For example: https://github.com/googlefonts/fontations/blob/d4651c5e31e81dbddac66163de66ab7fdf84488c/skrifa/src/metrics.rs#L304